mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt

```diff
@@ -134,25 +140,40 @@ abstract class AbstractBaseActivity : AppCompatActivity(), CoroutineScope {
     private fun timestampNeedsReauth(ts: Long) =
         ts == 0L || SystemClock.elapsedRealtime() - ts > AUTHENTICATION_VALIDITY_PERIOD
 
-    @CallSuper
-    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if (requestCode == SCREEN_LOCK_REQUEST_CODE) {
-            if (resultCode == RESULT_OK) {
-                lastAuthenticationTimestamp = SystemClock.elapsedRealtime()
-            } else if (getPrefs().getScreenLockMode(this) == ScreenLockMode.KioskMode) {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-                    finishAndRemoveTask()
-                } else {
-                    finish()
-                }
-            }
+    private inner class AuthPrompt : BiometricPrompt.AuthenticationCallback() {
+        private val contentView = findViewById<View>(R.id.activity_content)
+        private val prompt = BiometricPrompt(this@AbstractBaseActivity, Dispatchers.Main.asExecutor(), this)
+
+        fun authenticate() {
+            val info = BiometricPrompt.PromptInfo.Builder()
+                .setTitle(getString(R.string.app_name))
+                .setDescription(getString(R.string.screen_lock_unlock_screen_description))
```

Comment:
The description could be changed to "Unlock to access settings" if kiosk mode is enabled.

Reply:
```diff
diff --git a/mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt b/mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt
index 6b5d6843..457f8ed5 100644
--- a/mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt
@@ -145,9 +145,15 @@ abstract class AbstractBaseActivity : AppCompatActivity(), CoroutineScope {
         private val prompt = BiometricPrompt(this@AbstractBaseActivity, Dispatchers.Main.asExecutor(), this)
 
         fun authenticate() {
+            val description = if (getPrefs().getScreenLockMode(this@AbstractBaseActivity) ==
+                ScreenLockMode.KioskMode) {
+                getString(R.string.screen_lock_unlock_preferences_description)
+            } else {
+                getString(R.string.screen_lock_unlock_screen_description)
+            }
             val info = BiometricPrompt.PromptInfo.Builder()
                 .setTitle(getString(R.string.app_name))
-                .setDescription(getString(R.string.screen_lock_unlock_screen_description))
+                .setDescription(description)
                 .setDeviceCredentialAllowed(true)
                 .build()
             contentView.isInvisible = true
diff --git a/mobile/src/main/res/values/strings.xml b/mobile/src/main/res/values/strings.xml
index 92e04d98..4fe95f50 100644
--- a/mobile/src/main/res/values/strings.xml
+++ b/mobile/src/main/res/values/strings.xml
@@ -78,6 +78,7 @@
     <string name="settings_screen_lock">Screen lock</string>
     <string name="settings_screen_lock_summary">Lock openHAB access with Android screen lock or fingerprint. Voice widgets can always be used.</string>
     <string name="screen_lock_unlock_screen_description">Unlock to access openHAB</string>
+    <string name="screen_lock_unlock_preferences_description">Unlock to access settings</string>
     <string name="settings_screen_lock_off">Off</string>
     <string name="settings_screen_lock_kiosk">Kiosk mode</string>
     <string name="settings_screen_lock_on">On</string>

```